--- a/sys/hps_io.sv
+++ b/sys/hps_io.sv
@@ -216,6 +216,18 @@
 
 /////////////////////////////////////////////////////////
 
+reg  [7:0] kbd_data;
+reg        kbd_we;
+wire [8:0] kbd_data_host;
+reg        kbd_rd;
+reg  [7:0] mouse_data;
+reg        mouse_we;
+wire [8:0] mouse_data_host;
+reg        mouse_rd;
+generate if(!PS2DIV) begin
+ assign kbd_data_host=9'd0;
+ assign mouse_data_host=9'd0;
+end endgenerate
 wire [15:0] vc_dout;
 video_calc video_calc
 (
@@ -559,10 +571,6 @@
 			end
 		end
 
-		reg  [7:0] kbd_data;
-		reg        kbd_we;
-		wire [8:0] kbd_data_host;
-		reg        kbd_rd;
 
 		ps2_device keyboard
 		(
@@ -582,10 +590,6 @@
 			.rd(kbd_rd)
 		);
 
-		reg  [7:0] mouse_data;
-		reg        mouse_we;
-		wire [8:0] mouse_data_host;
-		reg        mouse_rd;
 
 		ps2_device mouse
 		(
