--- a/rtl/vendor/jt5205/hdl/jt5205_adpcm.v
+++ b/rtl/derived/jt5205_adpcm.v
@@ -72,7 +72,7 @@
     if( rst ) begin
         // sound fades away after a rst but the rest level must be -2
         // otherwise noises can be heard (e.g. intro scene of Double Dragon)
-        if( sound>12'd0 || sound < -12'd2 )
+        if( sound>12'sd0 || sound < -12'sd2 )
             sound <= sound >>> 1;
         else
             sound <= -12'd2;
